### ISSUE TITLE
Uploads hardcodes the content type to octet stream

### ISSUE THREAD

**Issue by pm-fawkes** [contentful/contentful-management.java#92]
The ModuleUploads.java hard codes the content type for uploads to octet stream, which means that anything loaded via this mechanism cannot be loaded by browsers correctly.

This was also a bug in the JS version of the CMA.

**Comment by Khaledgarbaya** [comment]
Hi @pm-fawkes,
This is not a bug, the direct file upload endpoint only accepts `octet-stream`. Mainly for security reasons.
Best,
Khaled

**Comment by pm-fawkes** [comment]
Not sure why this is a security issues - since the same bug was raised on the JS library and has been fixed!!!

If you wont' allow the content type to correctly specified on an upload then surely you should honour the content type specified on the asset when it is requested from images.contentful.com/....  currently it sends everything uploaded directly out with a content type of octet-stream.   If I upload via the UI or from a public URL then the content type served with the asset is correct.

**Comment by mariobodemann** [comment]
Hello,

maybe let us start over again. I'm Mario currently maintainer of the Java SDKs. @Khaledgarbaya is the maintainer of the JS SDKs. Since your questions was about a bug in the JS SDK, I asked to shed some light on the issue...

Now let me rephrase on what I think your question is: You are wondering why [Uploads uses OCTET_STREAM_CONTENT_TYPE](https://github.com/contentful/contentful-management.java/blob/master/src/main/java/com/contentful/java/cma/ModuleUploads.java#L74) for its uploads, since this might be a problem on the receiving side: You don't want it to be hard coded when you receive the asset from CDA.

I just created an image upload on Contentful, using the Java CMA SDK. It is an image and still used the OCTET content type in the upload module. If you request it now through CDA, you recieve a response like

```
{
  "fields": {
    "title": "title",
    "description": "locale",
    "file": {
      "url": "//images.contentful.com/arqlnkt58eul/1vg4AtlXLmomAmUeIAsouy/4f32ac1532ff55965cd08720ecc3af75/filename",
      "details": {
        "size": 3034,
        "image": {
          "width": 96,
          "height": 96
        }
      },
      "fileName": "filename",
      "contentType": "image/jpg"
    }
  }
}
```

As you can see, the content type of the asset is `image/jpg`. So the octet stream is only used on upload, since the upload endpoint requires it. You set the content type on the asset once you connect it to the upload.

I hope this could shed even more light on the question at hand...

**Comment by Khaledgarbaya** [comment]
Hey @pm-fawkes to add to what Mario said, there was a PR to make the content type configurable that slipped through and reverted later because the integration tests caught the error.

I hope the issue is a bit clear for now, otherwise please feel free to ask more questions

**Comment by pm-fawkes** [comment]
"You set the content type on the asset once you connect it to the upload."

We have done that - setting the content type on the asset after it is uploaded (perhaps my explanation was not clear enough) - but when we request the asset, it is still sent with a content-type of octet/stream.

**Comment by mariobodemann** [comment]
Hello @pm-fawkes,

That is is rather unfortunate, but it means that the asset got created or updated with  the wrong content type.

You can [set the content type of an asset](https://github.com/contentful/contentful-management.java/blob/master/src/main/java/com/contentful/java/cma/model/CMAAssetFile.java#L130) directly using the SDK. 

The hard coded part you were referring to is in a completely different module and had nothing to do with the described behaviour.

If you used the above method and updated an existing or created a new asset and published it, and still are recieving the asset with a wrong content type, please feel free to share an example with us, so we can investigate further.

Any source code leading to this behaviours and the resulting asset would be perfect to help you further.

 (Also if you feel you'd rather not share those details publicly, feel free to create a [support ticket](http://support.contentful.com))

Greetings,
Mario

**Comment by mariobodemann** [comment]
I will close this issue for now.

I hope we could clarify some issues here and please let us know if there is something else we can help you with. Either by reopening this issue, creating a new one or talking to our support.

Greetings,
Mario
